cores/esp8266/Schedule.cpp

```diff
@@ -80,22 +86,27 @@ bool schedule_function (const std::function<void(void)>& fn)
     return true;
 }
 
-bool schedule_recurrent_function_us (const std::function<bool(void)>& fn, uint32_t repeat_us)
+bool schedule_recurrent_function_us(const std::function<bool(void)>& fn, uint32_t repeat_us,
+    const std::atomic<bool>* wakeupToken)
 {
     assert(repeat_us < decltype(recurrent_fn_t::callNow)::neverExpires); //~26800000us (26.8s)
 
-    esp8266::InterruptLock lockAllInterruptsInThisScope;
+    if (!fn)
+        return false;
 
     recurrent_fn_t* item = new (std::nothrow) recurrent_fn_t(repeat_us);
     if (!item)
         return false;
 
     item->mFunc = fn;
+    item->mNext = nullptr;
+    item->wakeupToken = wakeupToken;
```

Comment:
What happens in this assignment? I'm not quite sure, but cppreference says at the top that std::atomic is not copyable.

Reply:
Here he;s just copying a pointer to the std::atomic, so that should be legal.

Reply:
Std::atomic & is out, std::function callback is in. Appears to be faster, even.